File: app/src/main/java/com/esbati/keivan/persiancalendar/Features/CalendarPage/CalendarFragment.java
package com.esbati.keivan.persiancalendar.Features.CalendarPage;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.widget.GridLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.esbati.keivan.persiancalendar.Components.SoundManager;
import com.esbati.keivan.persiancalendar.Features.Home.HomeFragment;
import com.esbati.keivan.persiancalendar.POJOs.CalendarDay;
import com.esbati.keivan.persiancalendar.R;
import com.esbati.keivan.persiancalendar.Repository.Repository;

/**
 * Created by asus on 11/17/2016.
 */

public class CalendarFragment extends Fragment {

    private final static String EXTRA_YEAR = "extra_year";
    private final static String EXTRA_MONTH = "extra_month";

    private int mYear;
    private int mMonth;

    private RecyclerView mRecyclerView;
    private RecyclerView.Adapter mAdapter;
    private RecyclerView.LayoutManager mLayoutManager;


    public void refreshCalendar(int newYear){
        mYear = newYear;

        mAdapter = new CalendarAdapter(mYear, mMonth, Repository.INSTANCE.prepareDays(mYear, mMonth));
        ((CalendarAdapter) mAdapter).setOnCalendarClickListener(new CalendarAdapter.OnCalendarClickListener() {
            @Override
            public void onCalendarClick(CalendarDay day) {
                SoundManager.getInstance().playSound(day.mDayNo);
                ((HomeFragment) getParentFragment()).showDate(day, true, false);
            }
        });
        mRecyclerView.setAdapter(mAdapter);
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        if(getArguments() != null){
            mYear = getArguments().getInt(EXTRA_YEAR);
            mMonth = getArguments().getInt(EXTRA_MONTH);
        }
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.fragment_calendar, container, false);

        setupView(rootView);
        return rootView;
    }

    public void setupView(View rootView){
        mRecyclerView = (RecyclerView)rootView.findViewById(R.id.list);
        //mRecyclerView.setNestedScrollingEnabled(false);

        mAdapter = new CalendarAdapter(mYear, mMonth, Repository.INSTANCE.prepareDays(mYear, mMonth));
        ((CalendarAdapter) mAdapter).setOnCalendarClickListener(new CalendarAdapter.OnCalendarClickListener() {
            @Override
            public void onCalendarClick(CalendarDay day) {
                SoundManager.getInstance().playSound(day.mDayNo);
                ((HomeFragment) getParentFragment()).showDate(day, true, false);
            }
        });
        mRecyclerView.setAdapter(mAdapter);

        mLayoutManager = new GridLayoutManager(getActivity(), 7, GridLayoutManager.VERTICAL, false);
        mRecyclerView.setLayoutManager(mLayoutManager);
    }

    public static CalendarFragment newInstance(int year, int month) {

        Bundle args = new Bundle();
        args.putInt(EXTRA_YEAR, year);
        args.putInt(EXTRA_MONTH, month);

        CalendarFragment fragment = new CalendarFragment();
        fragment.setArguments(args);
        return fragment;
    }
}

File: app/src/main/java/com/esbati/keivan/persiancalendar/features/calendarPage/CalendarFragment.kt
package com.esbati.keivan.persiancalendar.features.calendarPage

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.VERTICAL
import com.esbati.keivan.persiancalendar.R
import com.esbati.keivan.persiancalendar.components.SoundManager
import com.esbati.keivan.persiancalendar.components.locate
import com.esbati.keivan.persiancalendar.features.home.HomeFragment
import com.esbati.keivan.persiancalendar.pojos.CalendarDay
import com.esbati.keivan.persiancalendar.repository.Repository
import com.esbati.keivan.persiancalendar.utils.bindView

class CalendarFragment: Fragment() {

    private val mYear by lazy { arguments!!.get(EXTRA_YEAR) as Int }
    private val mMonth by lazy { arguments!!.get(EXTRA_MONTH) as Int }
    private val repository: Repository by locate()

    private val mRecyclerView: RecyclerView by bindView(R.id.list)
    private val mAdapter by lazy {
        CalendarAdapter(mYear, mMonth, repository.getDays(mYear, mMonth)).apply {
            onCalendarClickListener = object: CalendarAdapter.OnCalendarClickListener {
                override fun onCalendarClick(day: CalendarDay) {
                    SoundManager.playSound(day.mDay)
                    (parentFragment as HomeFragment).showDate(day, true)
                }
            }
        }
    }

    companion object {

        private const val EXTRA_YEAR = "extra_year"
        private const val EXTRA_MONTH = "extra_month"

        fun newInstance(year: Int, month: Int) = CalendarFragment().apply {
                arguments = Bundle().apply {
                    putInt(EXTRA_YEAR, year)
                    putInt(EXTRA_MONTH, month)
                }
            }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_calendar, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        mRecyclerView.layoutManager = androidx.recyclerview.widget.GridLayoutManager(activity, 7, VERTICAL, false)
        mRecyclerView.adapter = mAdapter
    }

    fun refreshCalendar() {
        mAdapter.refresh(repository.getDays(mYear, mMonth))
    }
}